server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -82,6 +84,11 @@ public ClusterState execute(ClusterState currentState, AddColumnRequest request)
             return currentState;
         }
 
+        // We cannot use normalizedColumns for building response since it may contain existing columns:
+        // parents of a sub-column or already existing reference, taken from the cluster state.
+        List<ColumnIdent> newColumnIdents = new ArrayList<>();
+        Consumer<Reference> addNewColumn = (ref) -> newColumnIdents.add(ref.column());
+
```

Comment:
Couldn't we do something like `addedColumns = request.references().stream().filter(ref -> currentTable.getReference(ref.column()) == null).toList();` to avoid having to change the `createMapping` function?